### ISSUE TITLE
Android : Not allowed to load local resource: file:///android_asset/www/index.html?auth_token

### ISSUE THREAD

**Issue by alain-andre** [lynndylanhurley/ng-token-auth#273]
Hi there,
I'm building an ionic app and I'm using ng-token-auth. Everything is fine in a browser, but when I emulate on android, I got an error : 

```batch
I/chromium( 2252): [INFO:CONSOLE(0)] "Not allowed to load local resource: file:///android_asset/www/index.html?auth_token=kUDlceeLBDmauQ2nN_KzqQ&blank=true&client_id=zoIcXXXXolEkpN0IxM9utA&config=&expiry=1459541228&uid=1727512764150849#/app/signin", source: file:///android_asset/www/index.html#/app/signin (0)
```
Here is my back-end code:  

`BASE_API_URL` depends if the app is launched on locahost or not. 

```javascript
$authProvider
  .configure({
    apiUrl: BASE_API_URL,
    storage: 'localStorage',
    forceHardRedirect: true,
    omniauthWindowType: window.cordova == undefined ? 'sameWindow' : 'inAppBrowser',
    authProviderPaths: {
      facebook: '/auth/facebook'
    }
  })
```

Any idea ? does it have to do something with [webview](http://stackoverflow.com/questions/19997146/kitkat-kills-not-allowed-to-load-local-resource-file-android-asset-webkit-a#tab-top) ? Or is it the Content-Security-Policy ? I'm stuck.

**Comment by angelxmoreno** [comment]
Have you gotten a chance to test this on a device? I have successfully implemented ng-token-auth on an Android device using Ionic. I have not tested it on an emulator though.

**Comment by alain-andre** [comment]
@angelxmoreno not yet. We are testing it on emulators for the moment. I will have a try on a real device asap and I'll tell you.

**Comment by alain-andre** [comment]
Hi @angelxmoreno, I have a JS error : 

```javascript
I/chromium(23010): [INFO:CONSOLE(49)] "Uncaught TypeError: Cannot read property 'open' of undefined", source: file:///android_asset/www/js/app.js (49)
```

Line *49* of *app.js* is `window.open = cordova.InAppBrowser.open;` in 
```javascript
if (window.cordova && window.cordova.plugins.Keyboard) {
  cordova.plugins.Keyboard.hideKeyboardAccessoryBar(true);
  cordova.plugins.Keyboard.disableScroll(true);
  window.open = cordova.InAppBrowser.open; // Tout fenêtre ouverte passera par le plugin inappbrowser de cordova.
}
```

**Comment by angelxmoreno** [comment]
@alain-andre I do not make use of `window.open = cordova.InAppBrowser.open` in my app.js because ng-token-auth spesifically refers to the open method in cordova.InAppBrowser. Please give that a try.

**Comment by alain-andre** [comment]
Hi @angelxmoreno we found the problem. It was because of the `omniauthWindowType: window.cordova == undefined ? 'sameWindow' : 'inAppBrowser'` line. Thanks for your help.

We changed it to `omniauthWindowType: 'inAppBrowser'` and now it works in emulation + real device. We just have an issue with the **ionic View** package that only shows a *redirecting....* message and do nothing. Even with `forceHardRedirect: true` it fails. I'm going to look at this in ionic forums.
